src/shapes/itext.class.js

```diff
@@ -487,24 +516,34 @@
      * @private
      * @param {CanvasRenderingContext2D} ctx
      */
-    _renderDragStartSelection: function (ctx) {
+    _renderDragSourceEffect: function (ctx) {
       if (this.__dragStartSelection) {
         this._renderSelection(
+          ctx,
           this.__dragStartSelection,
           this._getCursorBoundaries(this.__dragStartSelection.selectionStart, true),
-          ctx
         );
       }
     },
 
+    renderDropTargetEffect: function(e, withoutClearing) {
+      var dragSelection = this.getSelectionStartFromPointer(e);
+      if (withoutClearing) {
+        this._renderCursorAt(dragSelection);
+      }
+      else {
+        this.renderCursorAt(dragSelection);
+      }
+    },
+
     /**
      * Renders text selection
      * @private
      * @param {{ selectionStart: number, selectionEnd: number }} selection
      * @param {Object} boundaries Object with left/top/leftOffset/topOffset
      * @param {CanvasRenderingContext2D} ctx transformed context to draw on
      */
-    _renderSelection: function (selection, boundaries, ctx) {
+    _renderSelection: function (ctx, selection, boundaries, ctx) {
```

Comment:
@asturur why two `ctx` arguments?

Reply:
because i was sleepy. And also the second one was killing the first one, magic JS, even if undefined.